src/VisualStudio/Core/Test.Next/Services/VisualStudioDiagnosticAnalyzerExecutorTests.cs

```diff
@@ -120,6 +122,27 @@ End Sub
             }
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.RemoteHost)]
+        public async Task TestCancellationOnSessionWithSolution()
+        {
+            var code = @"class Test { void Method() { } }";
+
+            using (var workspace = CreateWorkspace(LanguageNames.CSharp, code))
+            {
+                var solution = workspace.CurrentSolution;
+                var solutionChecksum = await solution.State.GetChecksumAsync(CancellationToken.None);
+
+                var source = new CancellationTokenSource();
+                var connection = new MyConnection(source);
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(() => SessionWithSolution.CreateAsync(connection, solution, source.Token));
```

Comment:
❕  This should invoke the entry point for the error `RemoteHostClientExtensions.TryCreateSessionAsync` rather than using the inner helper methods from `SessionWithSolution`.

Reply:
that sure why. the bug is not there. but in CreateAsync.

Reply:
Testing of `TryCreateSessionAsync` would ensure that the complete sequence leading to the bug is corrected:

1. The connection is created
2. *Something happens*
3. The connection fails to be disposed

As-written, the test does not cover the first step of the scenario. I'll unblock because it may be substantially more work to write the more complete test.

Reply:
I believe I moved (2) down to CreateAsync. and nothing left in TryCreateSessionAsync. but if there is a bug, we still have Contract.Fail.